### PR TITLE
Show specificity in a css selector hover.

### PR DESCRIPTION
**Pull request by Ragnoroct**
This is a pull request to show the specificity score of a css selector on hover.

### CODE REVIEW

**Comment by Ragnoroct** [comment]
I'm kind of new to this. How do I reference https://github.com/Microsoft/vscode/issues/7299 with this pull request while I'm working on it?

**Comment by Ragnoroct** [comment]
I can fix the unit tests Saturday morning.

**Comment by Ragnoroct** [comment]
@aeschli I'm finished with this and all the tests pass. Do I need to do anything else to get this into vscode?

File: src/services/selectorPrinting.ts [hunk 1]
```diff
@@ -311,24 +310,59 @@ function unescape(content: string) {
...
+				case nodes.NodeType.ElementNameSelector:
+					//ignore universal selector
+					if (element.getText() === "*") {
+						break;
+					}
+					specificity[2] += 1;		//c
+					break;
+				case nodes.NodeType.PseudoSelector:
+					if (element.getText().match(/^::/)) {
+						specificity[2] += 1;	//c (speudo element)
```
Comment (kumarharsh) [inline src/services/selectorPrinting.ts]: minor correction: it's `pseudo`

File: src/services/selectorPrinting.ts [hunk 2]
```diff
@@ -311,24 +310,59 @@ function unescape(content: string) {
...
+					} else {
+						//ignore psuedo class NOT
+						if (element.getText().match(/^:not/i)) {
+							break;
+						}
+						specificity[1] += 1;	//b (speudo class)
```
Comment (kumarharsh) [inline src/services/selectorPrinting.ts]: Same here

Comment (kumarharsh) [inline src/services/selectorPrinting.ts]:
```suggestion
						specificity[1] += 1;	//b (pseudo class)
```

**Comment by Ragnoroct** [comment]
@kumarharsh Thank you. I've applied your suggested code changes to fix the typos.

**Comment by aeschli** [comment]
Very cool, thanks, @Ragnoroct !

**Comment by alicercedigital** [comment]
How can we hide this popup?

**Comment by aeschli** [comment]
You can turn off all css hovers, if you want:
```
  "[css]": {
    "editor.hover.enabled": false
  },
```

**Comment by artnez** [comment]
> You can turn off all css hovers, if you want:
> 
> ```
>   "[css]": {
>     "editor.hover.enabled": false
>   },
> ```

That turns off _all_ CSS hovers.

**Pull request merged**